### ISSUE TITLE
How to know customer received the tracking number or not ?

### ISSUE THREAD

**Issue by mohan-mandal-pattern** [allegro/allegro-api#8923]
I noticed in the app that we have added the tracking number to the order, but the customer cannot seem to view it on their end. Can you please let me know if there is an issue causing this, or if not, can you provide instructions on how the customer can view their tracking number?

**Comment by Lukasz-Zurek** [comment]
First of all - the customer should see it in his [purchase details](https://allegro.pl/moje-allegro/zakupy/kupione) and - if he has enabled notifications for shipment numbers - he should receive an e-mail. Please give me a trace-id of this request.

**Comment by mohan-mandal-pattern** [comment]
1. Are you referring to the order ID or something else? Could you please clarify?
2. We attempted to use the API endpoint to modify the order status. However, while the seller is able to view the status change, on the buyer side, the order status remains the same even after we added the tracking number and updated the fulfillment status to "SENT". Can you please identify the issue and explain why the order status is not changing on the buyer's side?

**Comment by Lukasz-Zurek** [comment]
1. I meant a trace-id of the request - we return it in the response headers. I would like to be sure if the tracking number has been added correctly. If yes, the customer should see it the [purchase details](https://allegro.pl/moje-allegro/zakupy/kupione) and track the shipment.
2. The order fulfillment status is only for the seller. It means that the customer/buyer won't see any status changes. He will see only tracking statuses. 

If the seller has enabled [subscription](https://allegro.pl/moje-allegro/sprzedaz/abonamenty), in the [Orders](https://allegro.pl/moje-allegro/sprzedaz/zamowienia/ustawienia) tab he can set automatic:

- change the status to “SENT”, when a tracking number(waybill) is added,
- sending a message to the buyer after the status is changed.

**Comment by mohan-mandal-pattern** [comment]
How can he see the tracking status? Can you please tell me?

**Comment by Lukasz-Zurek** [comment]
As a seller you can do it using [GET /order/carriers/{carrierId}/tracking?waybill={waybill}](https://developer.allegro.pl/documentation/#operation/getAllegroParcelTrackingUsingGET), you can read more [here](https://developer.allegro.pl/tutorials/how-to-manage-parcels-via-ship-with-allegro-ZM9YAyGKWTV#how-to-retrieve-parcels-statuses-history).

**Comment by mohan-mandal-pattern** [comment]
But How can the customer see the tracking status?

**Comment by Lukasz-Zurek** [comment]
The customer can track through [purchase details](https://allegro.pl/moje-allegro/zakupy/kupione) for every single order. [More information](https://help.allegro.com/sell/en/a/why-it-is-beneficial-and-how-to-provide-customers-with-a-tracking-number-AgbzE74VmH2).

**Comment by Lukasz-Zurek** [comment]
@mohan-mandal-pattern Please note that not every shipment can be tracked in the purchase details. If the seller adds "other" carrier - then we will not be able to track it so we will not provide it to the customer.

So if you have a specific case where you have added a tracking number, please give me more details.

**Comment by mohan-mandal-pattern** [comment]
Earlier we were using carrier-id as `OTHER` and passing the tracking number and our buyer sandbox account was not able to see the updating status of the shipment tracking.
We tried passing (after checking the shipment carrier list)FEDEX in the carrier-id with the tracking number in the sandbox and the buyer is still not able to view the shipment updation for the order.
- can the order shipment (waiting to be sent, in delivery ) status change in the sandbox, any way we can test that

Will it work the same way in the production environment as well?
 Thank you

**Comment by Lukasz-Zurek** [comment]
@mohan-mandal-pattern We are still checking for which couriers on the test environment you can check shipment status. I will be back with some details as soon as possible.

**Comment by mohan-mandal-pattern** [comment]
@Lukasz-Zurek do you have any updates to share with me?

**Comment by Lukasz-Zurek** [comment]
@mohan-mandal-pattern I am still waiting for some answers (an information which couriers can you track on sandbox) from other department. I will let you know in this thread soon.

**Comment by Lukasz-Zurek** [comment]
@mohan-mandal-pattern Unfortunately I do not have any good news, as we do not have such integrations (on the sandbox environment) that would simulate the process of sending and delivering a shipment in the form of statuses on Sandbox. This is why we don't return statuses on sandbox, in your orders.

As far as I know, the maximum that can be achieved is the status that the shipment is in preparation. For this purpose, you could for example, generate an Orlen Paczka label. But it would have to be done through the [Ship with Allegro](https://developer.allegro.pl/tutorials/how-to-manage-parcels-via-ship-with-allegro-ZM9YAyGKWTV) service.

So if you want to add a waybill to the order via the public API and then see the statuses, we don't have a process for that.

**Comment by mohan-mandal-pattern** [comment]
@Lukasz-Zurek Are you sure that the shipment tracking update has been integrated into the production account?

**Comment by Lukasz-Zurek** [comment]
@mohan-mandal-pattern Yes, it is and we are integrated with some specific couriers which we have mentioned in [this article](https://help.allegro.com/sell/en/a/carriers-integrated-with-allegro-yPWa9O2YGIL). But unfortunately it does not work on the test environment (sandbox).
